### PR TITLE
Remove reference to CommandBase and SubsystemBase

### PR DESCRIPTION
**Pull request by rzblue**
Companion to wpilibsuite/allwpilib#5392

### CODE REVIEW

File: source/docs/software/commandbased/command-based-changes.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ The overall structure of the command-based framework has remained largely the sa
...
+``Command`` (`Java <https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Command.html>`__, `C++ <https://github.wpilib.org/allwpilib/docs/release/cpp/classfrc2_1_1_command.html>`__) and ``Subsystem`` (`Java <https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Subsystem.html>`__, `C++ <https://github.wpilib.org/allwpilib/docs/release/cpp/classfrc2_1_1_subsystem.html>`__) are both now interfaces as opposed to abstract classes, allowing advanced users more potential flexibility.  ``CommandBase`` and ``SubsystemBase`` abstract base classes are still provided for convenience, but are not required.  For more information, see :doc:`commands` and :doc:`subsystems`. // TODO this isn't correct anymore
```
Comment (sciencewhiz) [inline source/docs/software/commandbased/command-based-changes.rst:19]: update per todo

**Review by sciencewhiz — COMMENTED** [review]
where applicable, the javadocs/doxygen links should point to development rather then release so they are correct for being the class vs interface. They'll be converted back to release once there is a 2024 release

File: source/docs/software/commandbased/command-based-changes.rst (hunk 1 above)
Comment (rzblue) [reply source/docs/software/commandbased/command-based-changes.rst:19]: That article will be deleted by #2287, i'll just remove my changes to the file.

**Review by sciencewhiz — APPROVED** [review]

**Comment by katzuv** [comment]
Should that now be (partially) reverted, after https://github.com/wpilibsuite/allwpilib/pull/5634?

**Pull request merged**